src/fit.jl

```diff
@@ -132,28 +139,44 @@ function train!(
         end
     end
 
+    if truncate_booster && estimator.early_stopping_round > 0 && results["best_iter"] > 0 # truncate_booster flag on AND early_stopping enabled
+        truncate_model!(estimator, results["best_iter"])
+    end
+
     # save the model in serialised form, in case we should be deepcopied or serialised elsewhere
     estimator.model = LGBM_BoosterSaveModelToString(estimator.booster)
 
     return results
+
 end
 # Old signature, pass through args
 function train!(
-    estimator::LGBMEstimator, tests_names::Vector{String}, verbosity::Integer, start_time::DateTime
+    estimator::LGBMEstimator, tests_names::Vector{String}, verbosity::Integer, start_time::DateTime;
+    truncate_booster::Bool=true,
 )
-    return train!(estimator, estimator.num_iterations, tests_names, verbosity, start_time)
+    return train!(estimator, estimator.num_iterations, tests_names, verbosity, start_time, truncate_booster=truncate_booster)
+end
+
+
+function truncate_model!(estimator::LGBMEstimator, best_iteration::Integer)
+    current_iteration = LGBM_BoosterGetCurrentIteration(estimator.booster)
+    times_to_rollback = current_iteration - best_iteration # current_iteration must be >= best_iteration
+    for _ in [1:times_to_rollback;]
```

Comment:
I think you can just iterate over the 1:n range syntax without wrapping it up into an array